File: java/src/test/java/swaglabs/pages/CheckoutPage.java
package swaglabs.pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;

public class CheckoutPage {

    private final Page page;
    public final Locator firstName;
    public final Locator lastName;
    public final Locator postalCode;
    public final Locator continueButton;
    public final Locator finishButton;
    public final Locator completeHeader;
    public final Locator backToProductsButton;
    public final Locator errorMessage;

    public CheckoutPage(Page page) {
        this.page = page;
        this.firstName = page.getByTestId("firstName");
        this.lastName = page.getByTestId("lastName");
        this.postalCode = page.getByTestId("postalCode");
        this.continueButton = page.getByTestId("continue");
        this.finishButton = page.getByTestId("finish");
        this.completeHeader = page.getByTestId("complete-header");
        this.backToProductsButton = page.getByTestId("back-to-products");
        this.errorMessage = page.getByTestId("error");
    }

    public void fillInfo(String first, String last, String zip) {
        firstName.fill(first);
        lastName.fill(last);
        postalCode.fill(zip);
        continueButton.click();
    }

    public void finish() {
        finishButton.click();
    }
}

File: python/pages/checkout_page.py
from playwright.sync_api import Page


class CheckoutPage:
    def __init__(self, page: Page) -> None:
        self.page = page
        self.first_name = page.get_by_test_id("firstName")
        self.last_name = page.get_by_test_id("lastName")
        self.postal_code = page.get_by_test_id("postalCode")
        self.continue_button = page.get_by_test_id("continue")
        self.finish_button = page.get_by_test_id("finish")
        self.complete_header = page.get_by_test_id("complete-header")
        self.back_to_products_button = page.get_by_test_id("back-to-products")
        self.error_message = page.get_by_test_id("error")

    def fill_info(self, first: str, last: str, zip_code: str) -> None:
        self.first_name.fill(first)
        self.last_name.fill(last)
        self.postal_code.fill(zip_code)
        self.continue_button.click()

    def finish(self) -> None:
        self.finish_button.click()
